Update engines for grunt-dojo2

**Enhancement**

We are not running automated testing against node 4 while `engines` in `package.json` supports node >= 4.0.0.

I tried running tests under node 4 but ran into a problem with remap-istanbul, which requires node >= 6. We should consider either moving our engine support from grunt-dojo2 to node 6 or find a way to test node 4 by backporting remap-istanbul to node 4 or skip using remap-istanbul when using node 4. Otherwise, we don't have an automated way notifying us when builds fail on node 4.

**Error**

```

/Users/pshannon/src/dojo2/grunt-dojo2/node_modules/remap-istanbul/bin/remap-istanbul.js:19
		let buffer = '';
		^^^

SyntaxError: Block-scoped declarations (let, const, function, class) not yet supported outside strict mode
    at exports.runInThisContext (vm.js:53:16)
    at Module._compile (module.js:373:25)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Function.Module.runMain (module.js:441:10)
    at startup (node.js:140:18)
    at node.js:1043:3
cat: stdout: Broken pipe
npm ERR! Test failed.  See above for more details.

Process finished with exit code 1
```

--- a/package.json
+++ b/package.json
@@ -3,10 +3,10 @@
   "version": "0.1.3-pre",
   "description": "A package of Grunt tasks to use with Dojo 2 Packages",
   "engines": {
-    "node": ">=4.0.0",
+    "node": ">=6.0.0",
     "npm": ">=3.0.0"
   },
-  "homepage": "http://dojotoolkit.org",
+  "homepage": "https://dojo.io",
   "keywords": [
     "dojo2",
     "gruntplugin"

